aptos-move/framework/aptos-stdlib/sources/math128.move

```diff
@@ -75,4 +143,48 @@ module aptos_std::math128 {
         let result = pow(10u128, 0u128);
         assert!(result == 1, 0);
     }
+
+    #[test]
+    public entry fun test_mul_div() {
```

Comment:
Can you some tests to show case how muldiv with casting reduces rounding errors?

Reply:
Done